src/core/core.scale.js

```diff
@@ -1205,28 +1205,55 @@ var Scale = Element.extend({
 		var scaleLabelFont = helpers.options._parseFont(scaleLabel);
 		var scaleLabelPadding = helpers.options.toPadding(scaleLabel.padding);
 		var halfLineHeight = scaleLabelFont.lineHeight / 2;
+		var scaleLabelAlign = scaleLabel.align;
 		var position = options.position;
 		var rotation = 0;
-		var scaleLabelX, scaleLabelY;
+		var scaleLabelX, scaleLabelY, textAlign;
+		var isReverse = me.options.ticks.reverse;
 
 		if (me.isHorizontal()) {
-			scaleLabelX = me.left + me.width / 2; // midpoint of the width
-			scaleLabelY = position === 'bottom'
-				? me.bottom - halfLineHeight - scaleLabelPadding.bottom
-				: me.top + halfLineHeight + scaleLabelPadding.top;
+			switch (scaleLabelAlign) {
+			case 'start':
+				scaleLabelX = me.left;
+				textAlign = isReverse ? 'right' : 'left';
+				break;
+			case 'end':
+				scaleLabelX = me.left + me.width;
+				textAlign = isReverse ? 'left' : 'right';
+				break;
+			default:
+				scaleLabelX = me.left + me.width / 2;
+				textAlign = 'center';
+			}
+			scaleLabelY = position === 'top'
+				? me.top + halfLineHeight + scaleLabelPadding.top
+				: me.bottom - halfLineHeight - scaleLabelPadding.bottom;
 		} else {
 			var isLeft = position === 'left';
+			var startsAtBottom = !isReverse;
 			scaleLabelX = isLeft
 				? me.left + halfLineHeight + scaleLabelPadding.top
 				: me.right - halfLineHeight - scaleLabelPadding.top;
-			scaleLabelY = me.top + me.height / 2;
+			switch (scaleLabelAlign) {
+			case 'start':
+				scaleLabelY = me.top + (startsAtBottom ? me.height : 0);
+				textAlign = (startsAtBottom ^ !isLeft) ? 'left' : 'right';
```

Comment:
`^` isn't that common, so it'd probably be clearer not to use it. How about `startsAtBottom == isLeft`? Same below. Also the parentheses probably aren't required

Reply:
But that wont be the same thing, `^` reverses the left side when the right side is true.

```
const startsAtBottom = true;
const isLeft = false;

startsAtBottom ^ !isLeft // falsey
startsAtBottom == isLeft // truthy
```

I removed the paranthesis

Reply:
done